### ISSUE TITLE
ProductAdmin meta textfields throwing errors on state change

### ISSUE THREAD

**Issue by jshimko** [reactioncommerce/reaction#2581]
Editing meta textfields in the ProductAdmin component is causes an issue with the field's state which throws an error in the console.

<img width="1679" alt="basic_reaction_product_and_reaction_ _working_copy__register-component-api_ _29_changed_files_" src="https://user-images.githubusercontent.com/3077956/28440267-7e3e3846-6d73-11e7-83c1-8994ed285008.png">

The state change that's throwing the error is here:

https://github.com/reactioncommerce/reaction/blob/development/imports/plugins/included/product-admin/client/containers/productAdminContainer.js#L48-L68

The updates appear to work (which is how it has gone unnoticed), but I suspect there's a decent chance something is broken somewhere that we're just not seeing on the surface.  Either way, the immutable `update` method being used there is deprecated anyway, so it should probably be replaced.  

That container should probably be consolidated into the `<ProductDetail/>` component that it wraps too.  It's a little too complicated to debug a component full of state and event handlers that all get passed to another component below it.  They could all be in one place and then only pass the data in from the container.  Perhaps that's out of scope for this issue, but it also may be one of those "while you're there..." kind of things.

### Steps to Reproduce the Behavior

- `reaction init -b development`
- log in as admin and go to product detail page
- change to edit mode
- click an edit pencil to open product admin panel
- expand "Details" tab and start typing in any meta text field
- note the errors shown above in your console after every keystroke

### Versions
```
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.4.0
```

**Comment by aaronjudd** [comment]
@jshimko @mikemurray assigned to you.

**Comment by zenweasel** [comment]
Closed via #2683
